draft-ietf-masque-connect-udp-listen.md

```diff
@@ -107,11 +107,38 @@ each HTTP Datagram (see {{fmt-dgram-uncomp}}), or registered via capsules and
 then compressed (see {{fmt-capsule-assign}}).
 
 When performing URI Template Expansion of the UDP Proxying template (see
-{{Section 3 of CONNECT-UDP}}), the client sets both the "target_host" and the
-"target_port" variables to the '\*' character (ASCII character 0x2A). Note that
-the '\*' character MUST be percent-encoded before sending, per {{Section 3.2.2
+{{Section 3 of CONNECT-UDP}}), the client follows the same template as
+CONNECT-UDP and sets the "target_host" and the "target_port" variables
+to one of its targets. It adds the connect-udp-bind header as specified in
+{{hdr}} to indicate that it prefers bind but would accept regular
+CONNECT-UDP to the target in case the proxy does not support it.
+If the proxy supports CONNECT-UDP Bind, it returns the connect-udp-bind
+response header value set to true and the proxy allocates context ID 0
+to the target and compressed datagrams {#compressed-operation} are transmitted
+between the client and proxy. On the other hand, if the proxy does not support
+bind, it sets the connect-udp-bind response header to false or omits it
+altogether.
+
+If the client intends to do CONNECT-UDP bind without a fallback to regular
+CONNECT-UDP, it sets both the "target_host" and the "target_port" variables to
+the '\*'  character (ASCII character 0x2A). Note that the '\*' character MUST
+be percent-encoded before sending, per {{Section 3.2.2
 of !TEMPLATE=RFC6570}}.
 
+## Mixed Mode Operation {#mixed-mode}
+
+Alternatively, the client can request mixed mode operation where the client is
+willing to accept non-bind CONNECT-UDP if bind is unavailable. In such an
+instance, the client provides the target_host and target_port of a single
+target as specfied in {{Section 3 of CONNECT-UDP}} with the addition of
+the connect-udp-bind header field as defined in {{hdr}}. If the proxy supports
+CONNECT-UDP Bind, it will return the connect-udp-bind response header value
+set to true and the proxy will request a closed context for said target using
+a COMPRESSION_ASSIGN capsule {{capsule-assign}}. If the connect-udp-bind
+header is unset false in the response, the client and proxy must operate in
+regular, non-bind CONNECT-UDP mode as stated in {{CONNECT-UDP}}.
+
+
```

Comment:
We need to modify ever instance of "context ID 0" in the doc now. I see one for example on line 157